Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -20,4 +20,25 @@ def determine_rpath_paths
     paths += homebrew_extra_library_paths
     paths.to_path_s
   end
+
+  def x11
+    xorg = Formula["linuxbrew/xorg/xorg"]
+    if xorg.installed?
+      xorg.recursive_dependencies.each do |d|
+        dep_formula = d.to_formula
+        append_path "PKG_CONFIG_LIBDIR", dep_formula.lib/"pkgconfig"
+        append_path "PKG_CONFIG_LIBDIR", dep_formula.share/"pkgconfig"
```

Comment:
My major concern here is that this partially defeats the purpose of superenv by dragging in a *huge* number of Homebrew dependencies, but at least it allows X11 formulae (like cairo) to build without accidentally using host deps.